csharp-mode.el

```diff
@@ -146,7 +146,7 @@
 
 (c-lang-defconst c-other-kwds
   csharp '("select" "from" "where" "join" "in" "on" "equals" "into"
-           "orderby" "descending" "group" "nameof"))
+           "orderby" "descending" "group" "nameof" "when"))
```

Comment:
If we are to support all LinQ keywords, I believe we are missing "let" in this list.

https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/let-clause

Reply:
Added in 82bcec9

Reply:
Yeah. That's much better, and we seemingly missed quite a few more than I noticed :laughing: